samza-core/src/main/java/org/apache/samza/execution/JobGraph.java

```diff
@@ -54,23 +55,49 @@
   private final Set<StreamEdge> sinks = new HashSet<>();
   private final Set<StreamEdge> intermediateStreams = new HashSet<>();
   private final Config config;
-  private final StreamGraph streamGraph;
+  private final JobGraphJsonGenerator jsonGenerator = new JobGraphJsonGenerator();
 
   /**
    * The JobGraph is only constructed by the {@link ExecutionPlanner}.
    * @param config Config
    */
-  /* package private */ JobGraph(StreamGraph streamGraph, Config config) {
-    this.streamGraph = streamGraph;
+  JobGraph(Config config) {
     this.config = config;
   }
 
   /**
+   * Returns the configs for single stage job, in the order of topologically sort.
+   * @return list of job configs
+   */
+  public List<JobConfig> getJobConfigs() {
```

Comment:
Minor: @Override for public interface methods and remove javadoc (it'll be inherited).